pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -264,6 +340,48 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 	return nil
 }
 
+// getDefaultGateway returns the default gateway address with lower priority using route and device
+// statedb tables and the provided default gateway configuration.
+func (r *NeighborReconciler) getDefaultGateway(defaultGateway *v2.DefaultGateway) (string, error) {
+	var defaultRoute netip.Prefix
+	switch defaultGateway.AddressFamily {
+	case "ipv4":
+		defaultRoute = ipv4Default
+	case "ipv6":
+		defaultRoute = ipv6Default
+	default:
+		return "", fmt.Errorf("invalid address family %s", defaultGateway.AddressFamily)
+	}
+	txn := r.DB.ReadTxn()
+	// get routes from statedb route table
+	// TODO: add RoutePrefixIndex Query to lookup routes by prefix
+	routes := r.routeTable.All(txn)
+	activeDefaultRoutes := []*tables.Route{}
+	for route := range routes {
+		// ignore routes that are not default routes or do not have a valid gateway
+		if !route.Gw.IsValid() || route.Dst != defaultRoute {
+			continue
+		}
+		dev, _, found := r.deviceTable.Get(txn, tables.DeviceIDIndex.Query(route.LinkIndex))
+		// ignore routes if the link through which it is reachable is not up
+		if !found || dev.OperStatus != "up" {
+			continue
+		}
+		if route.Gw.IsLinkLocalUnicast() {
+			r.logger.Warn(fmt.Sprintf("link local address %s is not supported for default gateway mode of bgp auto-discovery", route.Gw.String()))
```

Comment:
Please don't use formatted text with Warn-level message (it's ok to use it for Debug). Please use the log field properly.

Reply:
fixed